magic-modules-plugin/src/main/kotlin/io/labs/dotanuki/magicmodules/MagicModulesPlugin.kt

```diff
@@ -20,12 +20,10 @@ class MagicModulesPlugin : Plugin<Settings> {
     }
 
     private fun Settings.computeModulesAndPatchSettings() {
-
         val extension = registerPluginExtension()
-        val parsedStructure = ProjectStructureParser().parse(settingsDir)
-        val processedScripts = BuildScriptsProcessor.process(parsedStructure)
-
         gradle.settingsEvaluated {
+            val parsedStructure = ProjectStructureParser(extension).parse(settingsDir)
+            val processedScripts = BuildScriptsProcessor.process(parsedStructure)
```

Comment:
Could you share the practical reasons why the instances now are created per settings evaluation?

Reply:
The main reason is that plugin configurations are lazy. When we apply a plugin, the apply function is called at the same time and only default values are available in the configuration lifecycle. So, we have to use a lazy function as `doLast`, `settingsEvaluated`, in this case, to get custom values.

Reply:
Makes sense to me!